test/tests/functional/pbs_power_provisioning_sgi.py

```diff
@@ -146,9 +146,9 @@ def setup_sgi_api(self, script, perm=0o755):
             if host in done:
                 continue
             done.add(host)
-            dir = "/opt/sgi/ta"
-            dest = os.path.join(dir, "sgi_power_api.py")
-            self.server.du.run_cmd(host, "mkdir -p " + dir, sudo=True)
+            pwr_dir = "/opt/clmgr/power-service"
```

Comment:
You should use os.path.join() here as well and in revet_sgi_api(). Also, in __get_pmi_type() you still check for /opt/sgi but you do not do so here. Does this break backward compatibility?

Codacy also indicated one possible new issue that should be addressed.

Reply:
Thanks for the comment. I have used path join now.

In _get_pmi_type() I have maintained the backward compatibility but whereas in ptl since we are creating a fake script we need the API file at any one place. So I chose to have hpe path.

Reply:
Thank you for making the changes. Codacy seems to be on the fritz the past couple of days. When I looked at the previous report it cited one possible new issue involving a try/except block. Could you please check once the Codacy report has been generated?

Reply:
![image](https://user-images.githubusercontent.com/2702535/47171836-d0f02200-d2d7-11e8-8008-4da1efed9f5e.png)